src/unity/toolkits/supervised_learning/linear_regression_opt_interface.cpp

```diff
@@ -236,10 +236,10 @@ void linear_regression_opt_interface::compute_first_order_statistics(const
 /**
  * Compute the second order statistics
 */
-void linear_regression_opt_interface::compute_second_order_statistics(const
-    DenseVector& point, DenseMatrix& hessian, DenseVector& gradient, double&
-    function_value) {
-  
+void linear_regression_opt_interface::compute_second_order_statistics(
+    const ml_data& data, const DenseVector& point, DenseMatrix& hessian,
```

Comment:
Note that this `data` parameter shadows the class's `data` member. That's why there are no changes to the function itself. (We could change the parameter name, but honestly I think the class data members should be renamed to begin with m_ or perhaps end with _. But I didn't want to do that in this PR.)